File: fix_tables.js
const fs = require('fs');
const path = require('path');

const directoryPath = path.join(__dirname, 'src/app');

function walkDir(dir, callback) {
  fs.readdirSync(dir).forEach(f => {
    let dirPath = path.join(dir, f);
    let isDirectory = fs.statSync(dirPath).isDirectory();
    isDirectory ? walkDir(dirPath, callback) : callback(path.join(dir, f));
  });
}

function processFile(filePath) {
  if (!filePath.endsWith('.js') && !filePath.endsWith('.jsx')) return;
  
  let content = fs.readFileSync(filePath, 'utf8');
  let originalContent = content;

  // 1. Add fixed: 'right' to action column
  // Match standard action columns
  content = content.replace(/({\s*(?:title)?\s*:\s*['"`]操作['"`][^}]*key\s*:\s*['"`]action['"`][^}]*)}/g, (match, p1) => {
    if (p1.includes("fixed:")) return match;
    return p1 + ', fixed: \'right\' }';
  });

  // 2. Add scroll={{ x: 'max-content' }} to <Table ... />
  content = content.replace(/<Table\s+([^>]*?)>/g, (match, p1) => {
    let newAttr = p1;
    if (!newAttr.includes('scroll=')) {
      newAttr += ' scroll={{ x: \'max-content\' }}';
    }
    return `<Table ${newAttr}>`;
  });

  // 3. Add rowSelection to <Table ... />
  content = content.replace(/<Table\s+([^>]*?)>/g, (match, p1) => {
    let newAttr = p1;
    if (!newAttr.includes('rowSelection=')) {
      newAttr += ' rowSelection={{ type: \'checkbox\' }}';
    }
    return `<Table ${newAttr}>`;
  });

  if (content !== originalContent) {
    fs.writeFileSync(filePath, content, 'utf8');
    console.log('Updated:', filePath);
  }
}

walkDir(directoryPath, processFile);
console.log('Processing complete.');

File: fix_tables.py
import os
import re

directoryPath = os.path.join(os.path.dirname(__file__), 'src/app')

def process_file(file_path):
    if not file_path.endswith('.js') and not file_path.endswith('.jsx'):
        return

    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()
        
    original_content = content

    # 1. Add fixed: 'right' to action column
    # Look for { title: '操作', ..., key: 'action', ... }
    # Only applies if it doesn't already have fixed
    def replace_action(match):
        p1 = match.group(1)
        if 'fixed:' in p1:
            return match.group(0)
        return p1 + ", fixed: 'right' }"

    content = re.sub(r'({\s*(?:title)?[^}]*?[\'"`]操作[\'"`][^}]*?key\s*:\s*[\'"`]action[\'"`][^}]*)}', replace_action, content)

    # 2. Add scroll and rowSelection to <Table ... />
    def replace_table(match):
        p1 = match.group(1)
        new_attr = p1
        if 'scroll=' not in new_attr:
            new_attr += " scroll={{ x: 'max-content' }}"
        if 'rowSelection=' not in new_attr and 'dataSource=' in new_attr: 
            # Only add to data tables, sometimes <Table> wrapper exists without data but we want to be safe
            new_attr += " rowSelection={{ type: 'checkbox', fixed: 'left' }}"
        return f"<Table {new_attr}>"

    content = re.sub(r'<Table\s+([^>]*?)>', replace_table, content)

    if content != original_content:
        with open(file_path, 'w', encoding='utf-8') as f:
            f.write(content)
        print(f"Updated: {file_path}")

for root, _, files in os.walk(directoryPath):
    for f in files:
        process_file(os.path.join(root, f))
print('Processing complete.')
